packages/core/src/modules/proofs/formats/presentation-exchange/PresentationExchangeProofFormatService.ts

```diff
@@ -0,0 +1,309 @@
+
+    const attachment = this.getFormatData(
+      { options, presentation_definition: presentationDefinition },
+      format.attachmentId
+    )
+
+    return { attachment, format }
+  }
+
+  public async processRequest(agentContext: AgentContext, { attachment }: ProofFormatProcessOptions): Promise<void> {
+    const ps = this.presentationExchangeService(agentContext)
+    const { presentation_definition: presentationDefinition } = attachment.getDataAsJson<{
+      presentation_definition: PresentationDefinition
+    }>()
+    ps.validatePresentationDefinition(presentationDefinition)
+  }
+
+  public async acceptRequest(
+    agentContext: AgentContext,
+    { attachmentId, requestAttachment }: ProofFormatAcceptRequestOptions<PresentationExchangeProofFormat>
+  ): Promise<ProofFormatCreateReturn> {
+    const ps = this.presentationExchangeService(agentContext)
+
+    const format = new ProofFormatSpec({
+      format: PRESENTATION_EXCHANGE_PRESENTATION,
+      attachmentId,
+    })
+
+    const { presentation_definition: presentationDefinition, options } = requestAttachment.getDataAsJson<{
+      presentation_definition: PresentationDefinition
+      options?: { challenge?: string; domain?: string }
+    }>()
+
+    const { areRequirementsSatisfied, requirements } = await ps.selectCredentialsForRequest(
+      agentContext,
+      presentationDefinition
+    )
+
+    if (!areRequirementsSatisfied) {
+      throw new AriesFrameworkError('Requirements of the presentation definition could not be satifsied')
+    }
+
+    const credentials: InputDescriptorToCredentials = {}
+
+    requirements.forEach((r) => {
+      r.submissionEntry.forEach((r) => {
+        credentials[r.inputDescriptorId] = r.verifiableCredentials.map((c) => c.credential)
+      })
+    })
+
+    const presentation = await ps.createPresentation(agentContext, {
+      presentationDefinition,
+      credentialsForInputDescriptor: credentials,
+      challenge: options?.challenge,
+      domain: options?.domain,
+    })
+
+    if (presentation.verifiablePresentations.length > 1) {
+      throw new AriesFrameworkError('Invalid amount of verifiable presentations. Only one is allowed.')
+    }
+
+    const data = {
+      presentation_submission: presentation.presentationSubmission,
+      ...presentation.verifiablePresentations[0],
+    }
+
+    const attachment = this.getFormatData(data, format.attachmentId)
+
+    return { attachment, format }
+  }
+
+  public async processPresentation(
+    agentContext: AgentContext,
+    { requestAttachment, attachment }: ProofFormatProcessPresentationOptions
+  ): Promise<boolean> {
+    const ps = this.presentationExchangeService(agentContext)
+    const { presentation_definition: presentationDefinition } = requestAttachment.getDataAsJson<{
+      presentation_definition: PresentationDefinition
+    }>()
+    const presentation = attachment.getDataAsJson<
+      W3cVerifiablePresentation & { presentation_submission: PresentationSubmission }
```

Comment:
W3cVerifiablePresentation is incorrect I think, as it's a class, while we just take the object here. So the type will think it has `context` property, but it will actually have `@context` property. We need to use json transformer on it as well.